src/goTranspiler.ts

```diff
@@ -1649,13 +1647,13 @@ ${this.getIden(identation)}return nil`;
         const isVoid   = this.isInsideVoidFunction(node);
 
         const nodeEndsWithReturn = tryBodyEndsWithReturn && catchBodyEndsWithReturn && !isVoid;
-
+        const errorName = node.catchClause.variableDeclaration.name.escapedText;
         const catchBlock =`
     {		
         ${nodeEndsWithReturn ? 'ret__ :=' : ''} func(this *${this.className}) (ret_ interface{}) {
 		    defer func() {
-                if e := recover(); e != nil {
-                    if e == "break" {
+                if ${errorName} := recover(); ${errorName} != nil {
```

Comment:
@caoilainnl this change is causing a mismatch here 
<img width="798" height="274" alt="image" src="https://github.com/user-attachments/assets/940d03e8-e9a6-48cc-ac82-ae3f0973d8cb" />

Reply:
oh ok got it 🙏